pandas/_libs/parsers.pyx

```diff
@@ -2173,7 +2207,11 @@ def _concatenate_chunks(list chunks):
             result[name] = union_categoricals(arrs,
                                               sort_categories=sort_categories)
         else:
-            result[name] = np.concatenate(arrs)
+            if is_extension_array_dtype(dtype):
```

Comment:
what is this for?

Reply:
without this the `np.concatenate` call will return the `arrs` back to an `np.object_` array again.